go/dhcp/main.go

```diff
@@ -548,6 +595,25 @@ func (h *Interface) ServeDHCP(ctx context.Context, p dhcp.Packet, msgType dhcp.M
 
 				if Reply {
 
+					var info interface{}
+					var err error
+					Filter, found := GlobalFilterCache.Get(p.CHAddr().String() + "" + msgType.String())
```

Comment:
This and the code at https://github.com/inverse-inc/packetfence/pull/3784/files#diff-2a21322cf6c4918fdce4d3b8c0736ef4R449 are practicaly the same

Making this a function to remove code duplication would help

And it would also make this enormous function a bit more lean